libs/datasets/timeseries.py

```diff
@@ -996,6 +1001,60 @@ def drop_new_case_outliers(
     return new_timeseries
 
 
+def drop_tail_positivity_outliers(
+    timeseries: MultiRegionDataset,
+    zscore_threshold: float = 10.0,
+    diff_threshold_ratio: float = 0.01,
+) -> MultiRegionDataset:
+    """Identifies and drops outliers from the new case series.
+
+    Args:
+        timeseries: Timeseries.
+        zscore_threshold: Z-score threshold.  All new cases with a zscore greater than the
+            threshold will be removed.
+
+    Returns: timeseries with outliers removed from test_positivity_7d
+    """
+    df_copy = timeseries.timeseries.copy()
```

Comment:
Can you use timeseries.timeseries_wide_dates() instead of df_copy and grouped_df? Something like
```
test_pos = ts_wide_dates.loc[:, CommonFields.TEST_POSITIVITY_7D]
zscores = test_pos.apply(process_series, axis=1)
test_positivity_diffs = test_pos.diff(axis=1).abs()
```
?